catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -155,68 +171,102 @@ private boolean isNitfMimeType(String rawMimeType) {
     return false;
   }
 
-  private void process(List<ContentItem> contentItems) {
-    List<ContentItem> newContentItems = new ArrayList<>();
-    contentItems.forEach(contentItem -> process(contentItem, newContentItems));
-    contentItems.addAll(newContentItems);
+  public void setMaxSideLength(int maxSideLength) {
+    if (maxSideLength > 0) {
```

Comment:
❗️We should make these `private` fields `volatile` since they could be updated while being used by `process` or any other methods.

Reply:
Done